### ISSUE TITLE
Changing My Syncing direction (Client becomes server and vice versa)

### ISSUE THREAD

**Issue by arjunbharara** [Mimetis/Dotmim.Sync#1251]
It throws Out Dated Exceptions and I am handling it by 
agent.LocalOrchestrator.OnOutdated(outdated => 
{ 
 outdated.Action = OutdatedAction.Reinitialize; 
}); 
Here the problem is if my table has foreign key references and one of table in not included in my Setup(scope)
it throws me Foreign Key reference conflict while upserting the data.

**Comment by Mimetis** [comment]
A client can't become a server.

**Comment by arjunbharara** [comment]
Imagine you have two controllers in a high availability (HA) setup. Initially, one of these controllers is serving requests, and both controllers have their databases synchronized with each other. The database of the serving controller acts as the primary (server) database.

Now, if the primary controller goes down, the secondary controller takes over as the primary. At this point, you want the database of the new primary controller to also act as the server.

To achieve this, you are swapping the connection strings for the SyncAgent between the two controllers. However, during this process, you encounter a ClientOutDated exception. This exception occurs because the last client sync endpoint (2248) is earlier than the last server cleanup metadata (2268).

In other words, the new primary controller's database is trying to synchronize, but it has a last sync timestamp that is older than the last cleanup timestamp of the server's database. This indicates that some data may have been cleaned up on the server that the client database is not aware of, leading to the exception.

**Comment by Mimetis** [comment]
Unfortunately, **DMS** is not a _peer-to-peer_ sync framework...

When you define one database as a server, it will be always a server and can't become a client
And in the opposite, a client database can't become a server database.

Even if you resolve the last timestamp issue, I'm not sure that all the `GetChanges` methods (from both side) should fail at some point
The server `ID` is always `Guid.Empty` and you can't have two databases with such an ID.
